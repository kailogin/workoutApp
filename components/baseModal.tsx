import React, { ReactNode } from "react";
import { TouchableOpacity, View } from "react-native";
import { MaterialIcons } from "@expo/vector-icons";
import Modal from "react-native-modal";
import GestureRecognizer from "react-native-swipe-gestures";

import { Colors } from "../utils/colors";

interface BaseModalProps {
  children: ReactNode;
  isVisible: boolean;
  setIsVisible: React.Dispatch<React.SetStateAction<boolean>>;
}

export const BaseModal: React.FC<BaseModalProps> = ({
  children,
  isVisible,
  setIsVisible,
}: BaseModalProps) => {
  // --- RENDER ---

  return (
    <GestureRecognizer
      onSwipeDown={() => setIsVisible(false)}
      style={{
        flex: !isVisible ? 0 : 1,
      }}
    >
      <Modal
        isVisible={isVisible}
        style={{
          alignItems: "center",
          backgroundColor: Colors.BLACK,
          borderTopColor: Colors.ORANGE,
          borderTopWidth: 3,
          flex: 1,
          justifyContent: "center",
          marginBottom: 0,
          marginLeft: 0,
          marginRight: 0,
          marginTop: 90,
        }}
      >
        <TouchableOpacity
          onPress={() => setIsVisible(false)}
          style={{
            marginTop: 8,
          }}
        >
          <MaterialIcons name="close" size={24} color={Colors.WHITE} />
        </TouchableOpacity>

        <View style={{ flex: 1 }}>
          {/* TODO: Replace this with Form data and not the dummy text I added in addNewModal */}
          {children}
        </View>
      </Modal>
    </GestureRecognizer>
  );
};
