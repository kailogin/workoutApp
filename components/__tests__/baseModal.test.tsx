import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import { BaseModal } from "../baseModal";

const children = <h4>Test me</h4>;

test("Test BaseModal", () => {
  const snap = renderer
    .create(
      <BaseModal handleClose={() => null} isVisible={true}>
        {children}
      </BaseModal>
    )
    .toJSON();

  expect(snap).toMatchSnapshot();
});
