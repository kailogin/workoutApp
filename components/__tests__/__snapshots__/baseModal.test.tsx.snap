// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BaseModal 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onSwipeDown={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      deviceHeight={null}
      deviceWidth={null}
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      panResponderThreshold={4}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      statusBarTranslucent={false}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "borderTopColor": "#F2994A",
          "borderTopWidth": 3,
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "marginBottom": 0,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 90,
          "transform": Array [
            Object {
              "translateY": 1334,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Object {
            "flex": 1,
            "width": "90%",
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <h4>
            Test me
          </h4>
        </View>
        <View
          onLayout={[Function]}
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
          testID="toastAnimatedContainer"
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFF",
                "borderLeftColor": "#69C779",
                "borderLeftWidth": 5,
                "borderRadius": 6,
                "elevation": 2,
                "flexDirection": "row",
                "height": 60,
                "opacity": 1,
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 6,
                "width": 340,
              }
            }
            testID="toastTouchableContainer"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "justifyContent": "center",
                    "paddingHorizontal": 25,
                  },
                  undefined,
                ]
              }
              testID="toastContentContainer"
            />
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
